contracts/StandardBounty.sol

```diff
@@ -0,0 +1,319 @@
+          } else {
+              require(contributions[_contributionId].tokens[i].transfer(contributions[_contributionId].contributor, contributions[_contributionId].amounts[i]));
+          }
+      }
+  }
+
+  function fulfillBounty(address[] _fulfillers, uint[] _numerators, uint _denomenator, string _data)
+      public
+      validateNotTooManyFulfillments
+      sameLength(_fulfillers.length, _numerators.length)
+      sumToOne(_numerators, _denomenator)
+  {
+      fulfillments.push(Fulfillment(_fulfillers, _numerators, _denomenator, _data, false));
+
+      BountyFulfilled((fulfillments.length - 1));
+  }
+
+  function calculateFraction(uint _balance, uint _numerator, uint _denomenator)
+      public
+      pure
+      returns (uint newBalanceDiv)
+  {
+    require(_denomenator != 0);
+
+    //first multiplies by the numerator
+    uint newBalanceMult = _balance * _numerator;
+    require(newBalanceMult / _numerator == _balance);
+
+    //secondly divides by the denomenator
+    newBalanceDiv = newBalanceMult / _denomenator;
+  }
+
+  function acceptFulfillment(uint _fulfillmentId, StandardToken[] _payoutTokens, uint[] _tokenAmounts)
+      public
+      validateFulfillmentArrayIndex(_fulfillmentId)
+      sameLength(_payoutTokens.length, _tokenAmounts.length)
+      onlyController
+  {
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[_fulfillmentId];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
+          } else {
+            require(_payoutTokens[i].balanceOf(this) >= _tokenAmounts[i]);
+            require(_payoutTokens[i].transfer(fulfillment.fulfillers[j], calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator)));
+          }
+        }
+      }
+
+      FulfillmentAccepted(_fulfillmentId);
+  }
+
+  function acceptAndFulfill(address[] _fulfillers, uint[] _numerators, uint _denomenator, string _data, StandardToken[] _payoutTokens, uint[] _tokenAmounts)
+      public
+      sameLength(_payoutTokens.length, _tokenAmounts.length)
+      onlyController
+  {
+      fulfillBounty(_fulfillers, _numerators, _denomenator, _data);
+      hasPaidOut = true;
+      Fulfillment storage fulfillment = fulfillments[fulfillments.length - 1];
+      for (uint256 i = 0; i < _payoutTokens.length; i++){
+        for (uint256 j = 0; j < fulfillment.fulfillers.length; j++){
+          if (_payoutTokens[i] == address(0)){
+            require(this.balance >= _tokenAmounts[i]);
+            fulfillment.fulfillers[j].transfer(calculateFraction(_tokenAmounts[i], fulfillment.numerators[j], fulfillment.denomenator));
```

Comment:
since this can be technically zero... if balance was set to zero earlier. What does eth do on 0 transfer? curious.